qt/scientific_interfaces/Indirect/ContainerSubtraction.cpp

```diff
@@ -361,83 +233,36 @@ void ContainerSubtraction::updateCan() {
  */
 void ContainerSubtraction::plotPreview(int wsIndex) {
   m_uiForm.ppPreview->clear();
-
-  // Plot sample
-  if (m_csSampleWS) {
-    m_uiForm.ppPreview->addSpectrum(
-        "Sample", QString::fromStdString(m_sampleWorkspaceName), wsIndex,
-        Qt::black);
-  }
+  m_uiForm.ppPreview->setUpdatesEnabled(false);
 
   // Plot container
   if (m_csContainerWS) {
-    m_uiForm.ppPreview->addSpectrum(
-        "Container", QString::fromStdString(m_containerWorkspaceName), wsIndex,
-        Qt::red);
+    m_uiForm.ppPreview->addSpectrum("Container", m_transformedContainerWS,
+                                    wsIndex, Qt::red);
+  }
+
+  // Plot sample
+  if (m_csSampleWS) {
+    m_uiForm.ppPreview->addSpectrum("Sample", m_csSampleWS, wsIndex, Qt::black);
   }
 
   // Plot result
   if (!m_pythonExportWsName.empty()) {
     m_uiForm.ppPreview->addSpectrum(
         "Subtracted", QString::fromStdString(m_pythonExportWsName), wsIndex,
-        Qt::green);
+        Qt::blue);
```

Comment:
This changes the color of the result curve (container subtracted). It may be nice to inform the user about the meaning of the color (tooltip, documentation?). Subtraction is easy but it would make this change also very user-friendly.

Reply:
@reimundILL The legend wasn't enabled for the plot preview, I've now enabled it